File: app/src/main/java/io/tanjundang/study/knowledge/launchmode/OneActivity.java
package io.tanjundang.study.knowledge.launchmode;

import android.content.Context;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import io.tanjundang.study.R;

import static io.tanjundang.study.knowledge.launchmode.TwoActivity.SkipToActivityTwo;

public class OneActivity extends AppCompatActivity {

    String type;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_single_top_one);
        TextView tvTest = (TextView) findViewById(R.id.tvTest);
        tvTest.setText("当前Activity实例：" + this);
        type = getIntent().getStringExtra("type");
    }

    public void SkipToTwo(View v) {
        SkipToActivityTwo(this, type);
    }

    public static void SkipToActivityOne(Context context, String type) {
        Intent intent = new Intent(context, OneActivity.class);
        intent.putExtra("type", type);
        context.startActivity(intent);
    }

}

File: app/src/main/java/io/tanjundang/study/knowledge/launchmode/OneActivity.kt
package io.tanjundang.study.knowledge.launchmode

import android.content.Context
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView

import io.tanjundang.study.R
import io.tanjundang.study.base.BaseActivity
import io.tanjundang.study.knowledge.launchmode.TwoActivity.Companion.SkipToActivityTwo

class OneActivity : BaseActivity() {

    internal var type: String = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun initView() {
        setContentView(R.layout.activity_single_top_one)
        val tvTest = findViewById<View>(R.id.tvTest) as TextView
        tvTest.text = "当前Activity实例：$this"
    }

    override fun initData() {
        type = intent.getStringExtra("type")
    }

    fun SkipToTwo(v: View) {
        SkipToActivityTwo(this, type)
    }

    companion object {
        fun SkipToActivityOne(context: Context, type: String) {
            val intent = Intent(context, OneActivity::class.java)
            intent.putExtra("type", type)
            context.startActivity(intent)
        }
    }

}
